src/components/Auth/BaseStyleComponent.tsx

```diff
@@ -1,53 +1,105 @@
-import React, { FunctionComponent, useState } from "react";
-import { View, Image, Dimensions } from "react-native";
+import React, { FunctionComponent, useState, useEffect } from "react";
+import { View, Image, Dimensions, Keyboard, Platform } from "react-native";
 import WebView from "react-native-webview";
 import { useApp } from "../../hooks/app-context";
 import { ImageComponent } from "../shared/ImageComponent";
-import {IngressComponent} from "../shared/IngressComponent";
-import {NblocksButton} from "../shared/NblocksButton";
-import {NblocksModalComponent} from "../shared/NblocksModalComponent";
-import {SubTitleComponent} from "../shared/SubTitleComponent";
-import {TextComponent} from "../shared/TextComponent";
-
-const BaseStyleComponent:FunctionComponent<{title: string, subTitle: string}> = ({children, title, subTitle}) => {
-
-    const {name, logo, privacyPolicyUrl} = useApp();
-    const [showPrivacyPolicy, setShowPrivacyPolicy] = useState(false);
-
-    const renderPrivacyPolicy = () => {
-        if (privacyPolicyUrl) {
-            return (
-                <NblocksModalComponent mode="full" visible={showPrivacyPolicy} swipable={false} onCloseModal={() => setShowPrivacyPolicy(false)}>
-                    <WebView style={{flex: 1}} source={{uri: privacyPolicyUrl}}>
-                    </WebView>
-                    <NblocksButton title="Close" type="primary" onPress={() => setShowPrivacyPolicy(false)}></NblocksButton>
-                </NblocksModalComponent>
-                )
-        }
-    }
+import { IngressComponent } from "../shared/IngressComponent";
+import { NblocksButton } from "../shared/NblocksButton";
+import { NblocksModalComponent } from "../shared/NblocksModalComponent";
+import { SubTitleComponent } from "../shared/SubTitleComponent";
+import { TextComponent } from "../shared/TextComponent";
+
+const BaseStyleComponent: FunctionComponent<{
+  title: string;
+  subTitle: string;
+}> = ({ children, title, subTitle }) => {
+  const { name, logo, privacyPolicyUrl } = useApp();
+  const [showPrivacyPolicy, setShowPrivacyPolicy] = useState(false);
+  const [keyboardState, setKeyboardState] = useState(false);
+  // Set keyboard event listners
+  useEffect(() => {
+    let hidePrivacyPolicyCTA = Keyboard.addListener("keyboardDidShow", () => {
```

Comment:
Will this run on every property change causing a Keyboard.addListener to be called many times?

Reply:
useEffect runs function only after component mounts, with empty array it only runs after/on first render.